src/controllers/productController.ts

```diff
@@ -226,4 +226,32 @@ export const getProductsByBrandController = async (req: Request, res: Response):
       }
   }
 
+};
+
+export const getProductPriceAfterDiscount = async (req: Request, res: Response) => {
```

Comment:
I think we have to check if the discount exists, if yes, we must make sure that it's not expired before subtracting it from the original price.